File: Java/irrgarten/src/irrgarten/modelo/Weapon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten.modelo;

/**
 *
 * @author fnand
 */
public class Weapon extends CombatElement{
    
    ///////////////////////////////////////////////////////////////////
    ///////////////////          Atributos          ///////////////////
    ///////////////////////////////////////////////////////////////////
    
    
    
    ///////////////////////////////////////////////////////////////////
    /////////////////          Constructores          /////////////////
    ///////////////////////////////////////////////////////////////////
    
    
    
    public Weapon(float power, int uses) {
        super(power, uses);
    }

    ///////////////////////////////////////////////////////////////////
    //////////////////             Get's             //////////////////
    ///////////////////////////////////////////////////////////////////
    
    
    
    ///////////////////////////////////////////////////////////////////
    //////////////////             Set's             //////////////////
    ///////////////////////////////////////////////////////////////////
    
    
    
    ///////////////////////////////////////////////////////////////////
    //////////////////            Metodos            //////////////////
    ///////////////////////////////////////////////////////////////////
    
    public float attack (){
        return produceEffect();
    }

    
    @Override
    public String toString() {
        return super.toString();
    }
    
    
    
}

File: Irrgarten/weapon.rb
# frozen_string_literal: true
require_relative 'dice.rb'
require_relative 'combat_element'
module Irrgarten
  class Weapon < CombatElement
    attr_accessor :power, :uses

    def initialize (power, uses)
      super(power, uses)
    end

    def attack ()
      return produceEffect
    end

    def to_s
      "W["+super+"]"
    end

  end
end